lex.go

```diff
@@ -205,6 +204,15 @@ func (l *lexer) run() {
 	for l.state = lexRule; l.state != nil; {
 		l.state = l.state(l)
 	}
+	close(l.items)
+}
+
+func (l *lexer) close() {
```

Comment:
not sure `close` is the best naming

Reply:
Seems fine to me, alternate might be `stop()` or `done()` if you don't like `close()`?

Reply:
If you like it, I like it :-)